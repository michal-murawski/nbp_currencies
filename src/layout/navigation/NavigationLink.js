import React from 'react';
import { NavLink } from 'react-router-dom';
import { withStyles } from 'material-ui/styles';
import Button from 'material-ui/Button';
import PropTypes from 'prop-types';
import { NavigationLinkStyles } from './styles';

const NavigationLink = props => {
  const { classes, children, ...otherProps } = props;
  return (
    <Button>
      <NavLink className={classes.root} activeClassName={classes.active} {...otherProps}>
        {children}
      </NavLink>
    </Button>
  );
};

NavigationLink.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
    .isRequired,
};

export const NavigationLinkRaw = NavigationLink;

export default withStyles(NavigationLinkStyles)(NavigationLink);
