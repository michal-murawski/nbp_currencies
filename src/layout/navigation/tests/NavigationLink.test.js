import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { NavigationLinkRaw } from '../NavigationLink';

const mockClasses = { root: 'test_class ' };

describe('layout/NavigationLink', () => {
  it('should render properly the NavigationLink', () => {
    const ChildDiv = () => <div>Child test</div>;

    const wrapper = shallow(
      <NavigationLinkRaw to="/test" classes={mockClasses}>
        <ChildDiv />
      </NavigationLinkRaw>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
