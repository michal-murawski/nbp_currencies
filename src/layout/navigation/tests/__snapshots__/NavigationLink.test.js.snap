// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout/NavigationLink should render properly the NavigationLink 1`] = `
<WithStyles(Button)>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    className="test_class "
    to="/test"
  >
    <ChildDiv />
  </NavLink>
</WithStyles(Button)>
`;
